include/aws/common/array_list.h

```diff
@@ -10,6 +10,8 @@
 
 #include <stdlib.h>
 
+AWS_EXTERN_C_BEGIN
```

Comment:
Won’t this export symbols that don’t need to be exported?

Reply:
yeah, after reading the docs I'm still not sure.

I'll run tests to see if this results in any more symbols being exposed...

Reply:
extern just tells linker that symbol will be available later somewhere else. and extern C does the same thing, but tells c++ compiler not to mangle the symbols it is looking for.

From what i remember from c spec, function declarations have implicit extern on them, so extern c would only change mangling on them.
But global vars might change semantic a bit if they are now under extern. ex if you declared int in header before without extern each compilation unit would get its own copy, but with extern the variable will be shared between all compilation units. But i dont think we ever use variables in this way. 

extern itself does not affect visibility of the symbol in the final dynamic lib

Reply:
I ended up leaving the AWS_EXTERN_C_BEGIN where they were